src/tasks/AndroidAppBuilder/ApkBuilder.cs

```diff
@@ -98,19 +98,9 @@ public ApkBuilder(TaskLoggingHelper logger)
             throw new InvalidOperationException("Interpreter and AOT cannot be enabled at the same time");
         }
 
-        if (!string.IsNullOrEmpty(DiagnosticPorts))
+        if (!string.IsNullOrEmpty(DiagnosticPorts) && !Array.Exists(RuntimeComponents, runtimeComponent => string.Equals(runtimeComponent, "diagnostics_tracing", StringComparison.OrdinalIgnoreCase)))
         {
-            bool validDiagnosticsConfig = false;
-
-            if (string.IsNullOrEmpty(RuntimeComponents))
-                validDiagnosticsConfig = false;
-            else if (RuntimeComponents.Equals("*", StringComparison.OrdinalIgnoreCase))
-                validDiagnosticsConfig = true;
-            else if (RuntimeComponents.Contains("diagnostics_tracing", StringComparison.OrdinalIgnoreCase))
-                validDiagnosticsConfig = true;
-
-            if (!validDiagnosticsConfig)
-                throw new ArgumentException("Using DiagnosticPorts require diagnostics_tracing runtime component.");
+            throw new ArgumentException("Using DiagnosticPorts requires diagnostics_tracing runtime component.");
```

Comment:
nit: Adding a list of the runtime components that were specified could be useful when debugging issues.

Reply:
Done.